std-bits/table/src/main/java/org/enso/table/data/index/MultiValueIndex.java

```diff
@@ -87,19 +83,31 @@ public Table makeTable(Aggregator[] columns) {
         merged);
   }
 
-  private static Builder getBuilderForType(int type, int size) {
-    switch (type) {
-      case Storage.Type.BOOL:
-        return new BoolBuilder();
-      case Storage.Type.DOUBLE:
-        return NumericBuilder.createDoubleBuilder(size);
-      case Storage.Type.LONG:
-        return NumericBuilder.createLongBuilder(size);
-      case Storage.Type.STRING:
-        return new StringBuilder(size);
-      case Storage.Type.OBJECT:
-        return new ObjectBuilder(size);
+  public int[] makeOrderMap(int rowCount) {
+    if (this.locs.size() == 0) {
+      return new int[0];
     }
-    return new InferredBuilder(size);
+
+    int[] output = new int[rowCount];
+
+    int idx = 0;
+    for (List<Integer> rowIndexes : this.locs.values()) {
+      for (Integer rowIndex : rowIndexes) {
+        output[idx++] = rowIndex;
+      }
+    }
+
+    return output;
+  }
+
+  private static Builder getBuilderForType(int type, int size) {
+    return switch (type) {
+      case Storage.Type.BOOL -> new BoolBuilder();
+      case Storage.Type.DOUBLE -> NumericBuilder.createDoubleBuilder(size);
+      case Storage.Type.LONG -> NumericBuilder.createLongBuilder(size);
+      case Storage.Type.STRING -> new StringBuilder(size);
+      case Storage.Type.OBJECT -> new ObjectBuilder(size);
+      default -> new InferredBuilder(size);
```

Comment:
Is the `default` ever reachable?

Reply:
in theory not as not an enumeration for Type wanted a safety fall back.
One to revisit when we handle column data types.